lib/db/mysql.js

```diff
@@ -585,6 +585,60 @@ module.exports = function (log, error) {
       )
   }
 
+  MySql.prototype.multipleQueries = function (poolName, queries, finalQuery) {
+    var err
+    return this.getConnection(poolName)
+      .then(
+        function (connection) {
+          var results = []
+
+          return runQuery(0)
+            .then(
+              function (results) {
+                return results
+              },
+              function (e) {
+                err = e
+              }
+            )
+            .finally(
+              function () {
+                if (finalQuery) {
+                  return query(connection, finalQuery.sql, finalQuery.params)
+                    .finally(finish)
+                }
+
+                finish()
+
+                function finish () {
+                  connection.release()
+                  if (err) {
+                    throw err
```

Comment:
I *think* this is unnecessary but could easily be wrong.  `finally` is not allowed to change the return value of the promise, so if there was an error then `finally` will be called, its return value ignored, and the error transparently re-raised.  This might also simplify the logic above if you don't have to capture the error into `err` for later throwing.